hi-ml/src/health_ml/utils/common_utils.py

```diff
@@ -1,11 +1,458 @@
+    :param logger_fn: The logging function to use for logging this exception
+    """
+    logger_fn(f"{message} Exception: {ex}")
+    exc_type, exc_value, exc_traceback = sys.exc_info()
+    traceback.print_exception(exc_type, exc_value, exc_traceback, file=sys.stdout)
+
+
+def namespace_to_path(namespace: str, root: PathOrString = fixed_paths.repository_root_directory()) -> Path:
+    """
+    Given a namespace (in form A.B.C) and an optional root directory R, create a path R/A/B/C
+    :param namespace: Namespace to convert to path
+    :param root: Path to prefix (default is project root)
+    :return:
+    """""
+    return Path(root, *namespace.split("."))
+
+
+def path_to_namespace(path: Path, root: PathOrString = fixed_paths.repository_root_directory()) -> str:
+    """
+    Given a path (in form R/A/B/C) and an optional root directory R, create a namespace A.B.C.
+    If root is provided, then path must be a relative child to it.
+    :param path: Path to convert to namespace
+    :param root: Path prefix to remove from namespace (default is project root)
+    :return:
+    """
+    return ".".join([Path(x).stem for x in path.relative_to(root).parts])
+
+
+def remove_file_or_directory(pth: Path) -> None:
+    """
+    Remove a directory and its contents, or a file.
+    """
+    if pth.is_dir():
+        for child in pth.glob('*'):
+            if child.is_file():
+                child.unlink()
+            else:
+                remove_file_or_directory(child)
+        pth.rmdir()
+    elif pth.exists():
+        pth.unlink()
+
+
+@contextmanager
+def change_working_directory(path_or_str: PathOrString) -> Generator:
+    """
+    Context manager for changing the current working directory
+    """
+    new_path = Path(path_or_str).expanduser()
+    old_path = Path.cwd()
+    os.chdir(new_path)
+    yield
+    os.chdir(str(old_path))
+
 
 def _create_generator(seed: Optional[int] = None) -> torch.Generator:
+    """
+    Create Torch generator and sets seed with value if provided, or else with a random seed.
+
+    :param seed: Optional seed to set the Generator object with
+    :return: Torch Generator object
+    """
     generator = torch.Generator()
     if seed is None:
         seed = int(torch.empty((), dtype=torch.int64).random_().item())
     generator.manual_seed(seed)
     return generator
+
+
+def get_all_environment_files(project_root: Path) -> List[Path]:
+    """
+    Returns a list of all Conda environment files that should be used. This is firstly the InnerEye conda file,
+    and possibly a second environment.yml file that lives at the project root folder.
+    :param project_root: The root folder of the code that starts the present training run.
+    :return: A list with 1 or 2 entries that are conda environment files.
+    """
+    innereye_yaml = fixed_paths.get_environment_yaml_file()
+    project_yaml = project_root / fixed_paths.ENVIRONMENT_YAML_FILE_NAME
```

Comment:
to make the runner more flexible, we should think through this logic carefully in the next days.